### ISSUE TITLE
[SDL2] Xbox Controller D-Pad and rumble work on PC but not on Android

### ISSUE THREAD

**Issue by mechakotik** [libsdl-org/SDL#9562]
I am using SDL_GameControllerGetButton to get D-Pad state. On Android, it always returns 0 even when some D-Pad button is pressed, while working fine on PC. Gamepad rumble is also broken on Android but not on PC. I am using SDL 2.30.2 and Xbox Series controller. Here is my gamepad handling code if it will help: [link](https://github.com/mechakotik/tails-adventure/blob/develop/src/gamepad.cpp).

**Comment by jjensen** [comment]
Android only supports rumble via a wired gamepad connection, at least for Android 12 and Android 13. There are reports that Bluetooth rumble may work with an Xbox One controller connected via Bluetooth on Android 14.

**Comment by slouken** [comment]
The D-pad works fine here, and I confirmed that Bluetooth Xbox controllers don't have rumble on Android 14.

Can you try this APK and see if it works for you? It's built using the latest SDL3 code in main.
https://www.libsdl.org/tmp/testcontroller.apk

**Comment by mechakotik** [comment]
It fails to install as it days it's incompatible with my device. I have Pixel 7 Pro.

**Comment by slouken** [comment]
I've updated the APK, try again?

**Comment by mechakotik** [comment]
It works fine, looks like the problems are with my code.

**Comment by GeekSquirrel** [comment]
> The D-pad works fine here, and I confirmed that Bluetooth Xbox controllers don't have rumble on Android 14.
> 
> Can you try this APK and see if it works for you? It's built using the latest SDL3 code in main. https://www.libsdl.org/tmp/testcontroller.apk

I'm using citron emulator, which is built on top of SDL2. And on my Android 11 device, the sticks and dpad of a controller don't work. But If I use my Android 14 device, everything goes well. When I tried the testcontroller.apk. I got the same result, Android 14 works but Android 11 not.

I tested with 3 different controllers(1 is usb and 2 are bluetooth, including a xbox controller) and all of them got the same result.
